pkg/distributor/distributor.go

```diff
@@ -1728,6 +1728,24 @@ func (d *Distributor) sendWriteRequestToBackends(ctx context.Context, tenantID s
 		return ctx
 	}
 
+	var partitionsRequestContext func() context.Context
+	var partitionsRequestContextAndCancel func() (context.Context, context.CancelFunc)
+
+	if d.cfg.IngestStorageConfig.Migration.IngestStorageMaxWaitTime > 0 {
```

Comment:
The config option has been added under the "migration" block. Given we don't want this setting to be used unless during the migration, I would add an extra check to `DistributorSendToIngestersEnabled`:

```suggestion
	if d.cfg.IngestStorageConfig.DistributorSendToIngestersEnabled && d.cfg.IngestStorageConfig.Migration.IngestStorageMaxWaitTime > 0 {
```